bridges/YoutubeBridge.php

```diff
@@ -102,9 +136,105 @@ private function ytBridgeAddItem($vid, $title, $author, $desc, $time){
 		$item['uri'] = self::URI . 'watch?v=' . $vid;
 		$thumbnailUri = str_replace('/www.', '/img.', self::URI) . 'vi/' . $vid . '/0.jpg';
 		$item['content'] = '<a href="' . $item['uri'] . '"><img src="' . $thumbnailUri . '" /></a><br />' . $desc;
+		$this->ytBridgeAttachVideo($vid, $item);
 		$this->items[] = $item;
 	}
 
+	private function ytBridgeAttachVideo($vid, &$item) {
+		$attach_format = $this->getInput('attach_format');
+
+		if(empty($attach_format)) {
+			return;
+		}
+		$path = $this->ytBridgeDownloadVideo($vid, $attach_format);
+		if($path === null) {
+			return;
+		}
+
+		$path = realpath($path);
+		$mainDir = realpath(PATH_ROOT);
+
+		if(strpos($path, $mainDir) !== 0) {
+			error_log("Downloaded video in $path is not inside $mainDir");
+			return;
+		}
+
+		$urlPrefix = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ? 'https://' : 'http://';
+		$urlHost = (isset($_SERVER['HTTP_HOST'])) ? $_SERVER['HTTP_HOST'] : '';
+
+		// FIXME: Make base URL configurable instead of relying on host header and server vars
+		$url = $urlPrefix . $urlHost . substr($path, strlen($mainDir));
```

Comment:
I found it a bit odd that the cache dir is exposed under the document root by default but I saw that the PixivBridge also makes use of this already…